docs/design/Enhance-Generate-PodGroup-OwnerReferences-for-Normal-Pod.md

```diff
@@ -0,0 +1,116 @@
+# Enhance Generate PodGroup OwnerReferences for Normal Pod
+
+## Background
+Volcano scheduler can schedule normal workload such as single Pod, Deployment, CronJob, Statefulset etc., and other CRD workflow. For these normal pod, volcano controller will create a podgroup for these workload. When it is a single Pod, controller will set the pod as podgroup's `ownerReferences`, otherwise controller will set pod's `ownerReferences` as the podgroup's `ownerReferences`. 
+
+For issue [2143](https://github.com/volcano-sh/volcano/issues/2143) when deployment update, it will create new replicaset and volcano controller will create new podgroup for this replicaset. After several times update, the Deployment has many `Inqueue` podgroups. For example, it is not reasonable for one Deployment has four podgourps and three podgroups is in `Inqueue` status which will consume resources in `overcommit` plugin.
+
+<table>
+	<tr>
+	    <th>Deployment</th>
+	    <th>ReplicaSet</th>
+	    <th>PodGroup</th>  
+	</tr>
+	<tr>
+	    <td rowspan="4">deploy-with-volcano</td>
+	    <td>deploy-with-volcano-7bd985746b</td>
+	    <td>podgroup-4f5210d9-3cba-4a59-b61c-1b243f464708</td>
+	</tr>
+	<tr>
+	    <td>deploy-with-volcano-7f95dd7984</td>
+	    <td>podgroup-17fec14a-232c-4e1c-942e-e4d1289fc60b</td>
+	</tr>
+	<tr>
+	    <td>deploy-with-volcano-fd9dd69b6</td>
+	    <td>podgroup-5edaeb31-f16e-49c4-ac81-4aa331ea8f19</td>
+	</tr>
+	<tr>
+	    <td>deploy-with-volcano-995f85c44</td>
+	    <td>podgroup-0296521d-9dad-4d68-8abf-2d4b32f3bfbc</td>
+	</tr>
+</table>
+
+## Solution
+Some workload control pod directly such as Job, SparkApplication etc. For these workload, we can use the pod's `ownerReferences` as podgroup's `ownerReferences`. Some workload will control pod by other resources such as Deployment controls pod by Replicaset or Workflow controls pod by other resources.
+
+Volcano controller will introduce a configmap which stores GVK(group/version/kind) array.
+
+```
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: volcano-controller-configmap
+  namespace: volcano-system
+data:
+  volcano-controller.conf: |
+    podgroupLevel:
+    - apiversion: apps/v1                    
+      kind: deployment
+    - apiversion: batch/v1                    
+      kind: job
+    - apiversion: argoproj.io/v1alpha1                   
```

Comment:
If workflow engine (e.g. Argo) also set `ownerReference` to each steps and it's more reasonable to create PodGroup for each steps, how to handle such case?

Reply:
This configmap just a example. For workflow engine, each steps maybe have a self-governed resource(job/deployment). Users just need configure these resources GVK into configmap, then controller will search pod upstream `ownerReference` until find one which match the `podgroupLevel` rule.